src/main/java/macrobase/analysis/transform/BatchScoreFeatureTransform.java

```diff
@@ -28,8 +32,28 @@ public BatchScoreFeatureTransform(MacroBaseConf conf, Iterator<Datum> input) thr
     protected List<Datum> transform(List<Datum> data) {
         batchTrainScore.train(data);
         List<Datum> results = new ArrayList<>(data.size());
+        List<MetricsAndDensity> metricsAndDensities = null;
+        boolean dumpScores = false;
+        String scoreFile = conf.getString(MacroBaseConf.SCORE_DUMP_FILE_CONFIG_PARAM, "");
+        if (scoreFile != "") {
+            metricsAndDensities = new ArrayList<>(data.size());
+            dumpScores = true;
+        }
         for(Datum d : data) {
+            double score = batchTrainScore.score(d);
             results.add(new Datum(d, batchTrainScore.score(d)));
+            if (dumpScores) {
+                metricsAndDensities.add(new MetricsAndDensity(d.getMetrics(), score));
+            }
+        }
+        if (dumpScores) {
```

Comment:
Is it possible to do this inside a different utility class? @edgan8 had some ideas.